code/controllers/subsystem/timer.dm

```diff
@@ -236,7 +236,7 @@ SUBSYSTEM_DEF(timer)
 	if (flags & TIMER_UNIQUE)
 		SStimer.hashes[hash] = src
 	if (flags & TIMER_STOPPABLE)
-		SStimer.timer_id_dict["timerid[id]"] = src
+		SStimer.timer_id_dict["timerid"+num2text(id,8)] = src
```

Comment:
I also think that non-stoppable timers should have a junk/dummy ID, since they don't use them, and they shouldn't be used. I'd go with -1, or maybe even a string "BADTIMERID"